ecl/resource_ecl_provider_connectivity_tenant_connection_v2.go

```diff
@@ -0,0 +1,461 @@
+package ecl
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/nttcom/eclcloud"
+	"github.com/nttcom/eclcloud/ecl/provider_connectivity/v2/tenant_connections"
+)
+
+func resourceProviderConnectivityTenantConnectionV2() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceProviderConnectivityTenantConnectionV2Create,
+		Read:   resourceProviderConnectivityTenantConnectionV2Read,
+		Update: resourceProviderConnectivityTenantConnectionV2Update,
+		Delete: resourceProviderConnectivityTenantConnectionV2Delete,
+		Schema: map[string]*schema.Schema{
+			"name": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"description": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": &schema.Schema{
+				Type:     schema.TypeMap,
+				Optional: true,
+			},
+			"tenant_connection_request_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"device_type": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ValidateFunc: validation.StringInSlice([]string{
+					"ECL::Compute::Server",
+					"ECL::Baremetal::Server",
+					"ECL::VirtualNetworkAppliance::VSRX",
+				}, false),
+			},
+			"device_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"device_interface_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"attachment_opts_server": &schema.Schema{
+				Type:          schema.TypeList,
```

Comment:
I think it is not list.

Reply:
That's right. Let me think.

Reply:
Since the data type of `attachment_opts` is object, schema.`TypeMap` is suitable.
However, there is a Terraform specification that non-primitive types cannot be specified in Elem of TypeMap, and schema definition is not possible.
This problem is also known in the hashicorp community, but does not seem to have solved it yet.
https://github.com/hashicorp/terraform-plugin-sdk/issues/62
An alternative is to set the type of attachment_opts to `TypeList` and specify `MaxItems=1`.
https://github.com/hashicorp/terraform-plugin-sdk/issues/62#issuecomment-288364624